model/textparse/protobufparse.go

```diff
@@ -160,13 +160,41 @@ func (p *ProtobufParser) Histogram() ([]byte, *int64, *histogram.Histogram) {
 		sh.NegativeSpans[i].Offset = span.GetOffset()
 		sh.NegativeSpans[i].Length = span.GetLength()
 	}
+
+	if isFloatHistogram(h) {
+		fh := histogram.FloatHistogram{
+			Count:           h.GetSampleCountFloat(),
+			Sum:             sh.Sum,
+			ZeroThreshold:   sh.ZeroThreshold,
+			ZeroCount:       h.GetZeroCountFloat(),
+			Schema:          sh.Schema,
+			PositiveSpans:   sh.Copy().PositiveSpans,
+			PositiveBuckets: castInt64ToFloat64(sh.PositiveBuckets),
+			NegativeSpans:   sh.Copy().NegativeSpans,
+			NegativeBuckets: castInt64ToFloat64(sh.NegativeBuckets),
```

Comment:
We should not be using `sh` here because that is for non-float histograms. It takes the buckets from PositiveDelta and NegativeDelta for `sh`. But what we want here is buckets from h.PositiveCount and NegativeCount. Also, we should not create `sh` (i.e. the non-float histogram) in the first place if this is a float histogram.

So the flow should look more like to avoid unnecessary calculations.

```go
func (...) Histogram() (...) {
    // ...
    if isFloatHistogram(h) {
        // Create float histogram here
        // Return it
    }

    // Create integer histogram here and return it.
}
```